Automattic/wp-calypso
# Help Center: Happychat HE availability

#### Changes proposed in this Pull Request

- [x] Moved the [Happychat connection manager](https://github.com/Automattic/wp-calypso/blob/36fcd301d0fde08f136b6cb14ae4c451a3162d37/client/lib/happychat/connection.js) into a package.
- [x] Created a hook `useHappychatAuth` for getting user auth
- [x] Created a hook `useHappychatAvailable` that with the auth check to see if there are chat slots available
- [x] Used the last hook to set the Live chat availability

TODO: We have to consider if we need [skills and groups](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/happychat/utils.js#L17-L24), and if so implement them.

#### Testing instructions



* Checkout branch.
* Run `yarn test-client lib/happychat` and confirm that all tests succeed.
* Test that the chat still works



Related to #63815
Fixes #63815
